### DISCUSSION TITLE
Is there rack awareness in nebula graph?

### DISCUSSION THREAD

**Discussion by joruxs** [vesoft-inc/nebula#4256] [Q&A] [answered]
In for example Apache Cassandra there is a rack awareness concept (using the network topology strategy). The purpose of this is to "mark" in configuration which rack/chassi each server runs on.

For example, assume a 9-node cluster:

**Rack1**
server1
server4
server7

**Rack2**
server2
server5
server8

**Rack3**
server3
server6
server9

If we use this setup and then apply replication_factor=3 in our keyspace we can be sure that each piece of data is replicated among all three racks. This means that not only can we survive temporarily losing a server or two, we can survive losing all nodes in an entire rack/chassi.

Is this possible to achieve with Nebula Graph?

**Comment by xiajingchun** [comment, marked as answer]
Good point. There used to be a feature called "zone" in Nebula 2.6 very similar to this idea. Check it out https://docs.nebula-graph.io/2.6.2/7.data-security/5.zone/
However, the zone feature has been removed from 3.x since it's not mature enough. It might be brought back in the future but no timeline yet.

**Reply by wey-gu** [reply]
Yes, the term zone used to be introduced(and removed T_T) was to address the different levels of fault sets like the rack, chassis, etc.
